### DISCUSSION TITLE
Mis en place : qq améliorations possibles ?

### DISCUSSION THREAD

**Discussion by tioneb79** [ltoinel/amw#3] [General]
bonjour Ludovic, après plusieurs MAj de notre serveur nous avons réussi à mettre en place ton widget, ça marche très bien :)

il reste à mon avis à rendre ça un peu plus responsive, je vois que dans card.html on peut déclarer du code donc je vais m'en charger. J'ai vu que tu avais rajouté récemment un "truncate" sur le libellé du produit quand écran mobile, c'est déjà pas mal.

par contre c'est au niveau de la réponse du WS amz que je ne comprends pas.

exemple : avec les mots clé "3570k intel" dans la barre de recherche sur leur site, on obtient en premier le bon processeur :
https://www.amazon.fr/s?k=3570k+intel&i=computers 

Mais quand on passe par le WS et que l'on cherche la même chose on n'obtient pas ce résultat mais un autre processeur, approchant !
https://www.cowcotland.com/amazon/card?keyword=3570k+intel

tu penses pouvoir y faire qq chose ou c'est leur WS qui ne fonctionne pas comme attendu ?

Et merci pour ton boulot, en tous cas !

**Comment by tioneb79** [comment]
de même peux-tu gérer la non réponse ?
exemple :  https://www.cowcotland.com/amazon/card?id=b08m49ww51
ça tourne dans le vide... possible d'afficher qq chose malgré tout ?

**Reply by ltoinel** [reply]
Bonjour,

Un grand merci pour ce feedback auquel je répond avec un peu de retard, il faut absolument que je change mon email primaire sur Github.
Je vais regarder pour le résultat, c'est peut-être lié aux paramètres de recherche que j'utilise dans l'API qui offrent des réponses différentes entre les API et le portail Amazon .... 

Pour le résultat vide je regarde cela aussi. J'aimerai pouvoir en cible utiliser un algo de diffusion de contenu en fonction de la thématique de la page Web + A/B testing pour proposer des produits qui se vendent le plus.

Concernant les performances, je te conseille vivement d'utiliser Redis et d'activer le cache Redis qui change tout en terme de temps de réponses + limitation des quotas d'appels à Amazon.
Si tu as des propositions d'optimisation sur le fichier Card.html, n'hésite surtout pas ! Voir même on peut en proposer d'autres pour proposer des bannières de tailles différentes.

**Reply by ltoinel** [reply]
Concernant les résultats, j'affiche en priorité les produits en stock .... Ce qui pourrait expliquer la différence de réponses.

**Reply by ltoinel** [reply]
J'ai complétement revu le projet et modernisé les widgets depuis sur la version 3.
